nodeadm/internal/kubelet/image-credential-provider.go

```diff
@@ -54,29 +55,68 @@ func (k *kubelet) writeImageCredentialProviderConfig(cfg *api.NodeConfig) error
 	return util.WriteFileWithDir(imageCredentialProviderConfigPath, config, imageCredentialProviderPerm)
 }
 
-type imageCredentialProviderTemplateVars struct {
-	ConfigApiVersion   string
-	ProviderApiVersion string
-	EcrProviderName    string
-}
-
-func generateImageCredentialProviderConfig(cfg *api.NodeConfig, ecrCredentialProviderBinPath string) ([]byte, error) {
-	templateVars := imageCredentialProviderTemplateVars{
-		EcrProviderName: filepath.Base(ecrCredentialProviderBinPath),
+func generateImageCredentialProviderConfig(ecrCredentialProviderBinPath string) ([]byte, error) {
+	scheme := k8sruntime.NewScheme()
+	providerName := filepath.Base(ecrCredentialProviderBinPath)
+	defaultCacheDuration := &k8smetav1.Duration{Duration: 12 * time.Hour}
+	matchImages := []string{
+		"*.dkr.ecr.*.amazonaws.com",
+		"*.dkr-ecr.*.on.aws",
+		"*.dkr.ecr.*.amazonaws.com.cn",
+		"*.dkr-ecr.*.on.amazonwebservices.com.cn",
+		"*.dkr.ecr-fips.*.amazonaws.com",
+		"*.dkr-ecr-fips.*.on.aws",
+		"*.dkr.ecr.*.c2s.ic.gov",
+		"*.dkr.ecr.*.sc2s.sgov.gov",
+		"*.dkr.ecr.*.cloud.adc-e.uk",
+		"*.dkr.ecr.*.csp.hci.ic.gov",
 	}
 	kubeletVersion, err := GetKubeletVersion()
```

Comment:
this is now available in `cfg.Status.KubeletVersion`, so you should keep that param in this func